packages/reactive-element/src/reactive-element.ts

```diff
@@ -223,9 +223,18 @@ type AttributeMap = Map<string, PropertyKey>;
  * interface corresponding to the declared element properties.
  */
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
-export type PropertyValues<T = any> = keyof T extends PropertyKey
-  ? Map<keyof T, unknown>
-  : never;
+export interface PropertyValues<T = any> extends Map<keyof T, unknown> {
+  get<K extends keyof T>(k: K): T[K];
```

Comment:
How about `keys`, `values`, and `entries`? And I suppose `set` and `delete` would probably indicate something bad going on, but for completeness?

BTW here's a similar interface that I made in playground elements: https://github.com/google/playground-elements/blob/b896927494d63fe99850200c2b502d503b149e3e/src/playground-code-editor.ts#L17

Reply:
`has`, `delete`, and `keys` only use the key generic, so those don't ned any changes. I updated the value generic argument to `Map<keyof T, T[keyof T]>` so that `values`, are taken care of, and overrode `set()` to be generic itself.

Reply:
Cool! Would be nice to include test statements for `keys`, `values`, `entries`, `set`, and `delete` in the test file, too.

Reply:
Added a few of those, especially for `set()`, but I think there are diminishing returns on re-checking that the value generic is doing the right thing everywhere it's used in the base interface.